packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/queue/queue.ts

```diff
@@ -0,0 +1,389 @@
+ */
+export interface QueueExtensionProps {
+  /**
+   * The list of subscriptions for this service.
+   *
+   * @default none
+   */
+  readonly subscriptions?: ISubscribable[];
+
+  /**
+   * The user-provided default queue for this service.
+   *
+   * @default If the `eventsQueue` is not provided, a default SQS Queue is created for the service.
+   */
+  readonly eventsQueue?: sqs.IQueue;
+
+  /**
+   * The user-provided queue delay fields to configure auto scaling for the default queue.
+   */
+  readonly scaleOnLatency?: QueueAutoScalingOptions;
+}
+
+/**
+ * The topic-specific settings for creating the queue subscriptions.
+ */
+export interface TopicSubscriptionProps {
+  /**
+   * The SNS Topic to subscribe to.
+   */
+  readonly topic: sns.ITopic;
+
+  /**
+   * The object representing topic-specific queue and corresponding queue delay fields to configure auto scaling.
+   */
+  readonly topicSubscriptionQueue?: SubscriptionQueue;
+}
+
+/**
+ * `SubscriptionQueue` represents the subscription queue object which includes the topic-specific queue and its
+ * corresponding auto scaling fields.
+ */
+interface SubscriptionQueue {
+  /**
+   * The user-provided queue to subscribe to the given topic.
+   * If the `queue` is not provided, the default `eventsQueue` will subscribe to the given topic.
+   *
+   * @default none
+   */
+  readonly queue: sqs.IQueue;
+
+  /**
+   * The user-provided queue delay fields to configure auto scaling for the topic-specific queue.
+   *
+   * @default none
+   */
+  readonly scaleOnLatency?: QueueAutoScalingOptions;
+}
+
+/**
+ * Options for configuring SQS Queue auto scaling.
+ */
+interface QueueAutoScalingOptions {
+  /**
+   * Acceptable amount of time taken to process one message in the queue.
+   */
+  readonly messageProcessingTime: cdk.Duration;
+
+  /**
+   * Acceptable amount of time for processing a single message in the queue.
+   */
+  readonly acceptableLatency: cdk.Duration;
+}
+
+/**
+ * The `TopicSubscription` class represents an SNS Topic resource that can be subscribed to by the service queues.
+ */
+export class TopicSubscription implements ISubscribable {
+  public readonly topic: sns.ITopic;
+
+  public readonly queue?: sqs.IQueue;
+
+  private readonly subscriptionQueue?: SubscriptionQueue;
+
+  constructor(props: TopicSubscriptionProps) {
+    this.topic = props.topic;
+    this.subscriptionQueue = props.topicSubscriptionQueue;
+    this.queue = props.topicSubscriptionQueue?.queue;
+  }
+
+  /**
+   * This method sets up SNS Topic subscriptions for the SQS queue provided by the user. If a `queue` is not provided,
+   * the default `eventsQueue` subscribes to the given topic.
+   *
+   * @param extension `QueueExtension` added to the service
+   * @returns the queue subscribed to the given topic
+   */
+  public subscribe(extension: QueueExtension) : SubscriptionQueue {
+    let ret = {
+      queue: extension.eventsQueue,
+      queueDelay: extension.autoscalingOptions,
```

Comment:
Does this need to be renamed to:
```suggestion
      scaleOnLatency: extension.autoscalingOptions,
```